eZ/Publish/Core/REST/Server/Output/ValueObjectVisitor/VersionInfo.php

```diff
@@ -28,10 +29,37 @@ class VersionInfo extends ValueObjectVisitor
      */
     public function visit(Visitor $visitor, Generator $generator, $data)
     {
-        $versionInfo = $data;
-
         $generator->startHashElement('VersionInfo');
+        $this->visitVersionInfoAttributes($visitor, $generator, $data);
+        $generator->endHashElement('VersionInfo');
+    }
+
+    /**
+     * Maps the given version $status to a representative string.
+     *
+     * @param int $status
+     *
+     * @return string
+     */
+    protected function getStatusString($status)
+    {
+        switch ($status) {
+            case Values\Content\VersionInfo::STATUS_DRAFT:
+                return 'DRAFT';
+
+            case Values\Content\VersionInfo::STATUS_PUBLISHED:
+                return 'PUBLISHED';
+
+            case Values\Content\VersionInfo::STATUS_ARCHIVED:
+                return 'ARCHIVED';
+        }
+
+        // @todo FIXME: What exception to use?
+        throw new \Exception('Undefined version status: ' . $status);
```

Comment:
`\RuntimeException` because in this case it would be an internal software error

Reply:
Done in https://github.com/ezsystems/ezpublish-kernel/pull/2319/commits/ae1063ee6cad66c8df3ba1206aed747d0d9dc0e3.